### PR TITLE
hypervisor: mshv: Refactor emulator into separate module 

### PR DESCRIPTION
**Pull request by jinankjain**

### CODE REVIEW

**Review by liuw — COMMENTED** [review]
There is a typo in "hypervisor: mshv: Restrict depedency to x86". It should be "dependency"

**Comment by jinankjain** [comment]
> There is a typo in "hypervisor: mshv: Restrict depedency to x86". It should be "dependency".

Fixed.

**Review by liuw — APPROVED** [review]

File: hypervisor/src/mshv/mod.rs [hunk 1]
```diff
@@ -3,17 +3,16 @@
 // Copyright © 2020, Microsoft Corporation
 //
 
-use crate::arch::emulator::{PlatformEmulator, PlatformError};
+use crate::arch::emulator::PlatformEmulator;
```
Comment (rbradford) [inline hypervisor/src/mshv/mod.rs:7]: nit: Could you remove this whitespace line - whitespace lines means cargo fmt will not reorder the use statements correctly. If there are others in this PR please remove those too. Thanks!

Comment (jinankjain) [reply hypervisor/src/mshv/mod.rs]: Done

**Pull request merged**